packages/devextreme/js/__internal/scheduler/m_utils_time_zone.ts

```diff
@@ -55,17 +57,66 @@ const getDaylightOffset = (startDate, endDate) => new Date(startDate).getTimezon
 
 const getDaylightOffsetInMs = (startDate, endDate) => getDaylightOffset(startDate, endDate) * toMs('minute');
 
-const calculateTimezoneByValue = (timezone, date = new Date()) => {
-  // NOTE: This check could be removed. We don't support numerical timezones
-  if (typeof timezone === 'string') {
-    const dateUtc = createUTCDate(date);
-    return timeZoneDataUtils.getTimeZoneOffsetById(timezone, dateUtc.getTime());
+const calculateTimezoneByValue = (timeZone, date = new Date()): number | undefined => {
+  if (!timeZone) {
+    return undefined;
+  }
+
+  let offset;
+  try {
```

Comment:
Can we get rid of this ```try/catch``` block?

E.g:
```typescript
  const offset = getStringOffset(timeZone, date);

  if (!offset) {
    return undefined;
  }
```

Reply:
No. We can get invalid timezone as argument (for example "asia/1111") and the component must work after that.